src/DynamoCore/Graph/Workspaces/HomeWorkspaceModel.cs

```diff
@@ -565,12 +565,23 @@ private void NodeOnRequestSilenceNodeModifiedEvents(NodeModel node, bool value)
         public override void Clear()
         {
             WorkspaceClosed?.Invoke();
+            UndefineCBNFunctionDefinitions();
```

Comment:
So we have never undefined function definitions before loading a new workspace until now? Where they still around after a new home workspace was loaded?

Reply:
Yes, that is correct. There are two cores in the VM - one static and the other the runtime core. The runtime core was being reset every time a new workspace was opened but not the static core. I realized that if we reset the static core, we would also need to reload all libraries and packages again into the core, so I decided to simply undefine the function definitions from the static core instead.

Reply:
is this called both when opening a new workspace and creating a new workspace?

Reply:
When opening an existing workspace, this function is called in another place - when reading the JSON. There is another call to the same method in this PR.